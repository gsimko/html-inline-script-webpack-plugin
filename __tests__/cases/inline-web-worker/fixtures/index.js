// This file will be loaded as raw text as configured via webpack and asssets-loader
// eslint-disable-next-line import/extensions, import/no-unresolved
import workerSource from './worker.js?raw';

const blob = new Blob([
  workerSource
]);

// Obtain a blob URL reference to our worker 'file'.
const blobURL = window.URL.createObjectURL(blob);

const worker = new Worker(blobURL);

let result;

worker.onmessage = function (event) {
  if (!result) {
    result = document.createElement('div');
    result.setAttribute('id', 'result');

    document.body.append(result);
  }

  const record = document.createElement('pre');
  record.innerHTML = JSON.stringify(event.data);

  result.append(record);
};

window.addEventListener('load', () => {
  const button = document.getElementById('button');

  button.addEventListener('click', () => {
    worker.postMessage({ postMessage: true });
  });
});
