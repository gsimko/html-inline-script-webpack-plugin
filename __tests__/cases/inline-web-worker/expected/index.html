<!doctype html><html lang="en"><head><meta charset="utf-8"/><meta http-equiv="Content-Type" content="text/html; charset=utf-8"/><meta name="language" content="English"/><meta http-equiv="X-UA-Compatible" content="IE=edge"/><meta name="viewport" content="minimum-scale=1,initial-scale=1,width=device-width,shrink-to-fit=no"/><title>webpack test</title><script defer="defer">(()=>{"use strict";const e=new Blob(["onmessage = function (event) {\n  const workerResult = { timestamp: Date.now(), ...event.data };\n\n  workerResult.onmessage = true;\n\n  postMessage(workerResult);\n};\n"]),t=window.URL.createObjectURL(e),n=new Worker(t);let s;n.onmessage=function(e){s||(s=document.createElement("div"),s.setAttribute("id","result"),document.body.append(s));const t=document.createElement("pre");t.innerHTML=JSON.stringify(e.data),s.append(t)},window.addEventListener("load",(()=>{document.getElementById("button").addEventListener("click",(()=>{n.postMessage({postMessage:!0})}))}))})();</script></head><body><p>This is minimal code to demonstrate webpack usage</p><button id="button">Run Action</button></body></html>